src/test/java/org/broadinstitute/dsm/util/tools/TbosUPSKitTool.java

```diff
@@ -0,0 +1,510 @@
+        DDPInstance ddpInstance = DDPInstance.getDDPInstance("testboston");
+        String[] shortIds = participants.keySet().toArray(new String[0]);
+        int counter = 0;
+        for (; counter < shortIds.length; counter++) {
+            String participantShortId = shortIds[counter];
+            logger.info("starting on partcipant " + (counter+1) + "/" + shortIds.length);
+            ArrayList<Kit> kits = participants.get(participantShortId);
+            logger.info("Found " + kits.size() + " kits for the particpant in this file ");
+            String insertedShipmentId = null;
+            int kitCounter = 0;
+            for (; kitCounter < kits.size(); kitCounter++) {
+                Kit kit = kits.get(kitCounter);
+                Kit outboundKit = getDDBKitBasedOnKitLabel(SQL_SELECT_KIT_BY_KIT_LABEL + SQL_SELECT_OUTBOUND, kit.getKitLabel(), ddpInstance.getDdpInstanceId());
+                if (outboundKit == null) {
+                    logger.error("Outbound Kit found for " + kit.getKitLabel() + " is null!");
+                }
+                else {
+                    if (!kit.getGuid().equals(outboundKit.getGuid()) || !kit.getShortId().equals(outboundKit.getShortId())) {
+                        throw new RuntimeException("2 kits don't match for outbound for kitLabel " + kit.getKitLabel() + " file kit is for participant (" + kit.getShortId() + "," + kit.getGuid() + ") and DB kit is for participant (" + outboundKit.getShortId() + "," + outboundKit.getGuid() + ")");
+                    }
+                    insertedShipmentId = decideWhatToInsertForKit(kit, outboundKit, false);
+                    if (StringUtils.isNotBlank(insertedShipmentId)) {
+                        kit.setShipmentId(insertedShipmentId);
+                    }
+                }
+                kit.setPackageId(null);
+                Kit inboundKit = getDDBKitBasedOnKitLabel(SQL_SELECT_KIT_BY_KIT_LABEL + SQL_SELECT_INBOUND, kit.getKitLabel(), ddpInstance.getDdpInstanceId());
+                if (inboundKit == null) {
+                    logger.error("Inbound Kit found for " + kit.getKitLabel() + " is null!");
+                }
+                else {
+                    if (!kit.getGuid().equals(inboundKit.getGuid()) || !kit.getShortId().equals(inboundKit.getShortId())) {
+                        throw new RuntimeException("2 kits don't match for inbound for kitLabel " + kit.getKitLabel() + " file kit is for participant (" + kit.getShortId() + "," + kit.getGuid() + ") and DB kit is for participant (" + inboundKit.getShortId() + "," + inboundKit.getGuid() + ")");
+                    }
+                    decideWhatToInsertForKit(kit, inboundKit, true);
+                }
+                logger.info("Finished kit " + (kitCounter+1) + " / " + kits.size() + " for participant " + participantShortId);
+            }
+        }
+    }
+
+    public static String decideWhatToInsertForKit(Kit fileKit, Kit dbKit, boolean isRetrun) {
```

Comment:
This method decides what activities from the file is missing in DSM database, if a kit doesn't have any any activity, it inserts all that it is in the file